plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -649,6 +650,82 @@ public void cloneFromSCM(PomData pomData, String name, String version, File chec
         }
     }
 
+    /**
+     * In order to perform a clone, the following cli in this order will be runned in the checkoutDirectory
+     * <ul>
+     *     <li>git init</li>
+     *     <li>git remote add origin url</li>
+     *     <li>git fetch --depth=1 origin scmTag</li>
+     *     <li>git checkout FETCH_HEAD</li>
+     * </ul>
+     * @param connectionURL can have a format such scm:git:https://github.com/jenkinsci/mailer-plugin.git or https://github.com/jenkinsci/mailer-plugin.git
+     * @param scmTag the tag or sha1 to clone
+     * @param checkoutDirectory the directory where to clone the git repository
+     * @param shallowClone if <code>true</code> the fetch will use <code>--depth=1</code>
+     * @return success or not
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    public static boolean clone(String connectionURL, String scmTag, File checkoutDirectory, boolean shallowClone)
+            throws IOException, InterruptedException {
+
+        // maven scm is doing all of this, not sure why...
```

Comment:
“Historical reasons” probably. Forget about it.

Reply:
not really a question for you, but I need to remember/dig why we (Apache Maven team) did that 14yo ago.
I remember vaguely this handling the fact (similar to what svn was doing) the checkout/clone in an already existing local git clone directory and so only an update was needed (similar to what svn was doing) but this was kindy wrong assumption of old people used to svn and cvs.... :)

Reply:
LOL, this whole system was basically ported from CVS support!